gems/sorbet-runtime/test/types/edge_cases.rb

```diff
@@ -214,17 +214,62 @@ def self.post_hook; end
 
     assert_equal(
       [
-        :singleton_method_added,
-        :method_added,
-        # hook registration overwrites this method but it should
-        # still call the original defined above
         :singleton_method_added,
         :post_hook,
       ],
       klass.instance_variable_get(:@called)
     )
   end
 
+  it "forbids the user from bypassing sig validation by defining their own hooks" do
+    c = Class.new do
+      extend T::Sig
+
+      sig {params(name: Symbol).void}
+      def self.method_added(name)
+        super(name)
+      end
+
+      def self.singleton_method_added(name)
+        super(name)
+      end
+
+      sig {returns(Integer)}
+      def self.foo
+        "bad"
+      end
+    end
+    assert_raises(TypeError) do
+      c.foo
+    end
```

Comment:
Can we make a test like this that documents the behavior when the user didn't write `super`?

Reply:
If I take the well-behaved example, make the user-defined hooks not well-behaved by deleting super, and then assert that that raises, it causes other tests to be flaky.

Reply:
Resolved.